src/DataStructures/SliceVariables.hpp

```diff
@@ -78,13 +78,15 @@ Variables<TagsList> data_on_slice(const Variables<TagsList>& vars,
  *
  * \see data_on_slice
  */
-template <std::size_t VolumeDim, typename TagsList>
-void add_slice_to_data(const gsl::not_null<Variables<TagsList>*> volume_vars,
-                       const Variables<TagsList>& vars_on_slice,
-                       const Index<VolumeDim>& extents, const size_t sliced_dim,
-                       const size_t fixed_index) noexcept {
+template <std::size_t VolumeDim, typename VolumeTagsList,
+          typename SliceTagsList>
+void add_slice_to_data(
+    const gsl::not_null<Variables<VolumeTagsList>*> volume_vars,
+    const Variables<SliceTagsList>& vars_on_slice,
+    const Index<VolumeDim>& extents, const size_t sliced_dim,
+    const size_t fixed_index) noexcept {
   constexpr const size_t number_of_independent_components =
```

Comment:
This seems to defeat the purpose of having a `Variables` to begin with. I think if you want this type of wild-west behavior you should be operating on the `double*` (views) themselves. Requiring that the types of the tags are the same (not just the number of components) and that the completely prefix removed tags are the same I think is a reasonable amount of "loosing" (you have a typo in the commit message :) ). However, that should be enforced with a `static_assert` or two

Reply:
Added the static asserts, please check the fixup

Reply:
okay, do you think you should check that the prefix-removed tags are the same too? Since that's what you're stating as the reason for adding thsi feature?

Reply:
I'm not sure I understand your comment, one of the static asserts does that, or am I missing something?

Reply:
Yes, sorry I misread that. I was expecting `tmpl` comparisons like we've been doing. Go ahead and squash.